packages/ragbits-document-search/src/ragbits/document_search/_main.py

```diff
@@ -128,7 +128,27 @@ async def ingest_document(
         if document_processor is None:
             document_processor = self.document_processor_router.get_provider(document_meta)
 
-        elements = await document_processor.process(document_meta)
+        document_processor = self.document_processor_router.get_provider(document_meta)
+        return await document_processor.process(document_meta)
+
+    async def ingest(
+        self,
+        documents: Sequence[DocumentMeta | Document | Union[LocalFileSource, GCSSource]],
+        document_processor: Optional[BaseProvider] = None,
+    ) -> None:
+        """
+        Ingest multiple documents.
+
+        Args:
+            documents: The documents or metadata of the documents to ingest.
+            document_processor: The document processor to use. If not provided, the document processor will be
+                determined based on the document metadata.
+        """
+
+        elements = []
```

Comment:
```suggestion
        elements = await asyncio.gather(*[self._process_document(document, document_processor) for document in documents])
```

Reply:
I thought about it but opted for adding TODO instead, because each `self._process_document` may (and probably will) lead to network requests and uncontrolled paralyzation of a big number of network requests would not be a good idea and may even lead to failed requests.

Reply:
Ok, makes sense